app/src/main/java/org/simple/clinic/instantsearch/InstantSearchUpdate.kt

```diff
@@ -58,7 +58,12 @@ class InstantSearchUpdate : Update<InstantSearchModel, InstantSearchEvent, Insta
   private fun searchResultsLoaded(model: InstantSearchModel, event: SearchResultsLoaded): Next<InstantSearchModel, InstantSearchEffect> {
     if (!model.hasSearchQuery) return noChange()
 
-    return dispatch(ShowPatientSearchResults(event.patientsSearchResults, model.facility!!))
+    val effect = if (event.patientsSearchResults.isNotEmpty())
+      ShowPatientSearchResults(event.patientsSearchResults, model.facility!!)
+    else
+      ShowNoSearchResults
```

Comment:
Should this be an effect? Seems to be more appropriate as an implementation detail of the UI renderer which looks at the search results, right?

Reply:
Hopefully, this [comment ](https://github.com/simpledotorg/simple-android/pull/2257#discussion_r551903893) clarifies why I went with this usage.